### PR TITLE
Swift5 namespace changes

### PR DESCRIPTION
**Pull request by aymanbagabas**
Based on this discussion https://github.com/OpenAPITools/openapi-generator/pull/9583#pullrequestreview-669994076

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/swift5/modelObject.mustache [hunk 1]
```diff
@@ -76,16 +76,17 @@
     // Decodable protocol methods
 
     {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}}{{#objcCompatible}} required{{/objcCompatible}} init(from decoder: Decoder) throws {
-        let container = try decoder.container(keyedBy: String.self)
+        let container = try decoder.container(keyedBy: CodingKeys.self)
```
Comment (4brunu) [inline modules/openapi-generator/src/main/resources/swift5/modelObject.mustache:79]: @aymanbagabas could you please explain what the changes to the function `init(from decoder: Decoder) throws` do, or what do they fix?

**Comment by 4brunu** [comment]
@aymanbagabas Since this PR is big because of the sample projects, I opened https://github.com/aymanbagabas/openapi-generator/pull/16 to reduce the model changes, and make it easier to review this PR.

**Comment by 4brunu** [comment]
@aymanbagabas overall the PR looks good, I just would like to understand better the changes in the method `init(from decoder: Decoder) throws` before approving this PR.

**Comment by aymanbagabas** [comment]
> @aymanbagabas overall the PR looks good, I just would like to understand better the changes in the method `init(from decoder: Decoder) throws` before approving this PR.

Since we're using `Codable`, it's preferable to pair it with `CodingKeys` to encode/decode custom types as explained [here](https://developer.apple.com/documentation/foundation/archives_and_serialization/encoding_and_decoding_custom_types#2904058). This change should've been part of https://github.com/OpenAPITools/openapi-generator/commit/0f5e7d1e3cb9642bd46ec89ffe0bf66a922029f2 but somehow it slipped.

**Comment by 4brunu** [comment]
Thanks, looks good to me 👍

**Comment by wing328** [comment]
@aymanbagabas can you please resolve the merge conflicts when you've time?

**Comment by 4brunu** [comment]
Thanks for fixing the merge conflicts 👍

**Pull request merged**